lib/ts/wrappers/express.ts

```diff
@@ -0,0 +1,268 @@
+                if (!path.startsWith(supertokens.appInfo.apiBasePath)) {
+                    return next();
+                }
+
+                let requestRID = request.getHeaderValue(HEADER_RID);
+                if (requestRID !== undefined) {
+                    let matchedRecipe: RecipeModule | undefined = undefined;
+
+                    // we loop through all recipe modules to find the one with the matching rId
+                    for (let i = 0; i < supertokens.recipeModules.length; i++) {
+                        if (supertokens.recipeModules[i].getRecipeId() === requestRID) {
+                            matchedRecipe = supertokens.recipeModules[i];
+                            break;
+                        }
+                    }
+
+                    if (matchedRecipe === undefined) {
+                        // we could not find one, so we skip
+                        return next();
+                    }
+
+                    let id = matchedRecipe.returnAPIIdIfCanHandleRequest(path, method);
+                    if (id === undefined) {
+                        // the matched recipe doesn't handle this path and http method
+                        return next();
+                    }
+
+                    // give task to the matched recipe
+                    let requestHandled = await supertokens.handleAPI(
+                        matchedRecipe,
+                        id,
+                        request,
+                        response,
+                        path,
+                        method
+                    );
+                    if (!requestHandled) {
+                        return next();
+                    }
+                    return;
+                } else {
+                    // we loop through all recipe modules to find the one with the matching path and method
+                    for (let i = 0; i < supertokens.recipeModules.length; i++) {
+                        let id = supertokens.recipeModules[i].returnAPIIdIfCanHandleRequest(path, method);
+                        if (id !== undefined) {
+                            let requestHandled = await supertokens.handleAPI(
+                                supertokens.recipeModules[i],
+                                id,
+                                request,
+                                response,
+                                path,
+                                method
+                            );
+                            if (!requestHandled) {
+                                return next();
+                            }
+                            return;
+                        }
+                    }
+                    return next();
+                }
+            } catch (err) {
+                next(err);
+            }
+        };
+    },
+    errorHandler: () => {
+        return async (err: any, req: Request, res: Response, next: NextFunction) => {
```

Comment:
I believe that this function too can be extracted away into `supertokens.ts`

Reply:
👍🏼